### ISSUE TITLE
Fix `Test` warnings from `NumPy`

### ISSUE THREAD

**Issue by Alnusjaponica** [optuna/optuna#4938]
### Motivation

Replace deprecated casting with officially recommended way.

Currently, some unit tests rises following warnings: 
```
DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
```
This Issue aims to resolve them.
See some log of `Test` e.g. https://github.com/optuna/optuna/actions/runs/6267996976/job/17022123321.

### Suggestion

See https://numpy.org/doc/stable/release/1.25.0-notes.html#deprecations

**Comment by torotoki** [comment]
I will work on this.

**Comment by HideakiImamura** [comment]
Let me close this issue, since this was resolved in #4978.
